test-suite/src/snapshot_integration_tests.rs

```diff
@@ -0,0 +1,72 @@
+use spatialos_sdk::worker::{entity::Entity, snapshot::*, EntityId};
+use std::env;
+use std::f64::EPSILON;
+
+use crate::generated::improbable::*;
+use std::collections::BTreeMap;
+
+#[test]
+pub fn create_and_read_snapshot() {
+    let mut snapshot_path = env::temp_dir();
+    snapshot_path.push("test.snapshot");
+
+    let entity = get_test_entity().expect("Error");
+
+    {
+        let snapshot = SnapshotOutputStream::new(snapshot_path.clone()).expect("Error");
+        snapshot
+            .write_entity(EntityId::new(1), &entity)
+            .expect("Error");
+    }
+
+    {
+        let mut snapshot = SnapshotInputStream::new(snapshot_path).expect("Error");
+
+        assert!(snapshot.has_next());
+
+        let entity = snapshot.read_entity().expect("Error");
+
+        let position = entity.get::<Position>();
+        assert!(position.is_some());
+        let coords = &position.unwrap().coords;
+        assert!((10.0 - coords.x).abs() < EPSILON);
+        assert!((-10.0 - coords.y).abs() < EPSILON);
+        assert!((0.0 - coords.z).abs() < EPSILON);
```

Comment:
There's a nifty crate for doing approximately-equals comparisons like this: https://docs.rs/approx/0.3.1/approx/

Reply:
Oh nice! :smile: